aten/src/ATen/native/quantized/cpu/qnnpack_maxpool.cpp

```diff
@@ -0,0 +1,145 @@
+        input.ndimension() == 4,
+        "qnnpack_maxpool(): Expected input to be 4-dimensional: got ",
+        input.ndimension());
+    TORCH_CHECK(
+        kernel_size.size() == 2,
+        "qnnpack_maxpool(): Expected kernel_size to be 2-dimensional: got ",
+        kernel_size.size());
+    TORCH_CHECK(
+        stride.size() == 2,
+        "qnnpack_maxpool(): Expected stride to be 2-dimensional: got ",
+        stride.size());
+    TORCH_CHECK(
+        dilation.size() == 2,
+        "qnnpack_maxpool(): Expected dilation to be 2-dimensional: got ",
+        dilation.size());
+    TORCH_CHECK(
+        padding.size() == 2,
+        "qnnpack_maxpool(): Expected padding to be 2-dimensional: got ",
+        padding.size());
+
+    Tensor input_contig = input.contiguous();
+
+    initQNNPACK();
+    const auto scale = input_contig.q_scale();
+    const auto zero_point = input_contig.q_zero_point();
+    qnnp_operator_t qnnpackOperator_{nullptr};
+
+    int64_t padL = padding[0];
+    int64_t padT = padding[1];
+    int64_t kH = kernel_size[0];
+    int64_t kW = kernel_size[1];
+    int64_t strideH = stride[0];
+    int64_t strideW = stride[1];
+    int64_t dilationH = dilation[0];
+    int64_t dilationW = dilation[1];
+
+    TORCH_CHECK(
+        kH > 0 && kW > 0,
+        "qnnpack_maxpool(): kernel_size should be greater than zero.");
+    TORCH_CHECK(
+        strideH > 0 && strideW > 0,
+        "qnnpack_maxpool(): strides should be greater than zero.");
+
+    // Input is in NHWC format
+    int64_t batch_size = input_contig.size(0);
+    int64_t inH = input_contig.size(1);
+    int64_t inW = input_contig.size(2);
+    int64_t inC = input_contig.size(3);
+
+    const qnnp_status createStatus = qnnp_create_max_pooling2d_nhwc_u8(
+        padT /* input_padding_top */,
+        padL /* input_padding_right */,
+        padT /* input_padding_bottom */,
+        padL /* input_padding_left */,
+        kH /* pooling height */,
+        kW /* pooling width */,
+        strideH /* stride height */,
+        strideW /* stride width */,
+        dilationH /* dilation height */,
+        dilationW /* dilation width */,
+        inC /* input channels */,
+        std::numeric_limits<uint8_t>::min() /* output min */,
+        std::numeric_limits<uint8_t>::max() /* output max */,
+        0 /* flags */,
+        &qnnpackOperator_);
+    TORCH_INTERNAL_ASSERT(
+        createStatus == qnnp_status_success,
+        "failed to create QNNPACK MaxPool operator");
+    TORCH_INTERNAL_ASSERT(qnnpackOperator_ != nullptr);
+
+    int64_t outC = inC;
+    int64_t outH =
+        pooling_output_shape(inH, kH, padT, strideH, dilationH, false);
+    int64_t outW =
+        pooling_output_shape(inW, kW, padL, strideW, dilationW, false);
+
+    TORCH_CHECK(
+        outH > 0 && outW > 0,
+        "qnnpack_maxpool(): the resulting output Tensor size should be >= 0");
+
+    // NHWC output
+    std::vector<int64_t> outSizes{batch_size, outH, outW, outC};
```

Comment:
This is a memory allocation, that might be skippable if you pass `{batch_size, outH, outW, outC}` directly as an argument.  Want to try it?